server/api.go

```diff
@@ -621,15 +653,27 @@ func (a *API) oauthRedirectHandler(w http.ResponseWriter, r *http.Request) {
 
 	w.Header().Add("Content-Type", "text/html")
 	if mmUser.Id == a.p.GetBotUserID() {
-		connectionMessage := "The bot account has been connected"
-		_, _ = w.Write([]byte(fmt.Sprintf("<html><body><h1>%s</h1><p>You can close this window.</p></body></html>", connectionMessage)))
+		http.Redirect(w, r, a.p.GetURL()+"/account-connected?isBot", http.StatusSeeOther)
 		return
 	}
 
 	_, _ = a.p.updateAutomutingOnUserConnect(mmUserID)
 
+	switch {
+	case a.p.getConfiguration().SyncNotifications:
+		a.handleSyncNotificationsWelcomeMessage(originInfo[0], mmUserID, postID)
+		http.Redirect(w, r, a.p.GetURL()+"/account-connected", http.StatusSeeOther)
+		return
+	default:
+		a.handleDefaultWelcomeMessage(originInfo[0], mmUserID, postID, channelID)
+		http.Redirect(w, r, a.p.GetURL()+"/primary-platform", http.StatusSeeOther)
+		return
+	}
```

Comment:
Not, necessarily as part of this PR, but should we have checks here for "RemoteOnlySync" and NOT "SelectiveSync" since those won't require selecting a PrimaryPlatform.

Reply:
I will redirect to `/primary-platforn` only if SelectiveSync is true!